### DISCUSSION TITLE
[build fail]YOU_MIXED_DIFFERENT_NUMERIC_TYPES__YOU_NEED_TO_USE_THE_CAST_METHOD_OF_MATRIXBASE_TO_CAST_NUMERIC_TYPES_EXPLICITLY

### DISCUSSION THREAD

**Discussion by chunleili** [InteractiveComputerGraphics/SPlisHSPlasH#229] [Q&A] [answered]
Hi,

I want to use this lib as the third party in my project. But I encountered such Eigen error, which does not encountered in a standalone build of the splishsplash.

The related error message is like this:

>[build] E:\codes\vcpkg\installed\x64-windows\include\eigen3\Eigen\src/Core/AssignEvaluator.h(888,3): error C2338: YOU_MIXED_DIFFERENT_NUMERIC_TYPES__YOU_NEED_TO_USE_THE_CAST_METHOD_OF_MATRIXBASE_TO_CAST_NUMERIC_TYPES_EXPLICITLY [E:\Dev\zenos\zeno\build\projects\SPH\SPlisHSPlasH\Simulator\SimulatorBase.vcxproj]

I build it in Windows 10, VS2019 + VS Code with cmake

**Comment by janbender** [comment, marked as answer]
Does SPlisHSPlasH compiles as standalone library correctly? 
It is hard to say where this error comes from. Do you use the same Eigen library in your project and in the SPlisHSPlasH lib?

**Reply by chunleili** [reply]
There is no error if I build the splishsplash standalone. The splishsplash uses the eigen source files from the extern directory , but my project uses the eigen from the vcpkg.

**Reply by janbender** [reply]
This could already be the problem. Can you try to use the same Eigen lib in both libraries?

**Reply by janbender** [reply]
In CMake you can change the EIGEN3_INCLUDE_DIR to do this.

**Reply by chunleili** [reply]
Problem solved. Thanks for your help!

> This could already be the problem. Can you try to use the same Eigen lib in both libraries?
